src/Sylius/Component/Attribute/Model/AttributeValue.php

```diff
@@ -126,31 +145,117 @@ public function getName()
     /**
      * {@inheritdoc}
      */
-    public function getPresentation()
+    public function getType()
     {
         $this->assertAttributeIsSet();
 
-        return $this->attribute->getPresentation();
+        return $this->attribute->getType();
     }
 
     /**
-     * {@inheritdoc}
+     * @return boolean
      */
-    public function getType()
+    public function getBoolean()
     {
-        $this->assertAttributeIsSet();
+        return $this->boolean;
+    }
 
-        return $this->attribute->getType();
+    /**
+     * @param boolean $boolean
+     */
+    public function setBoolean($boolean)
+    {
+        $this->boolean = $boolean;
+    }
+
+    /**
+     * @return string
+     */
+    public function getText()
+    {
+        return $this->text;
+    }
+
+    /**
+     * @param string $text
+     */
+    public function setText($text)
+    {
+        $this->text = $text;
+    }
+
+    /**
+     * @return int
+     */
+    public function getInteger()
+    {
+        return $this->integer;
+    }
+
+    /**
+     * @param int $integer
+     */
+    public function setInteger($integer)
+    {
+        $this->integer = $integer;
+    }
+
+    /**
+     * @return float
+     */
+    public function getFloat()
+    {
+        return $this->float;
+    }
+
+    /**
+     * @param float $float
+     */
+    public function setFloat($float)
+    {
+        $this->float = $float;
+    }
+
+    /**
+     * @return \DateTime
+     */
+    public function getDateTime()
+    {
+        return $this->dateTime;
+    }
+
+    /**
+     * @param \DateTime $dateTime
+     */
+    public function setDateTime(\DateTime $dateTime)
+    {
+        $this->dateTime = $dateTime;
+    }
+
+    /**
+     * @return \DateTime
+     */
+    public function getDate()
+    {
+        return $this->date;
+    }
+
+    /**
+     * @param \DateTime $date
+     */
+    public function setDate(\DateTime $date)
+    {
+        $this->date = $date;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getConfiguration()
+    public function __toString()
```

Comment:
Isn't it just `$this->getValue()`?

Anyway, I suggest to move this method just after properties declaration so the magic methods will be always in one place.